wave_lang/kernel/wave/perf/benchmark_speculative_sampling.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2025 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import logging
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s - %(levelname)s - %(message)s",
+)
+logger = logging.getLogger(__name__)
+
+import argparse
+import importlib
+import json
+
+from pathlib import Path
+import sys
+from typing import Any, Dict, List
+
+import torch
+
+from wave_lang.kernel.lang.global_symbols import *
+from tests.kernel.wave.speculative_decode_test import testReferenceSpeculativeDecoding
+
+from wave_lang.kernel.wave.perf.utils import analyze_rpd_trace
+
+rpd_tracer_module = importlib.util.find_spec("rpdTracerControl")
+if rpd_tracer_module is None:
+    logger.error("rpdTracerControl not found. Profiling is disabled.")
+    sys.exit(1)
+else:
+    rpdTracerControl = importlib.import_module("rpdTracerControl").rpdTracerControl
+
+DEFAULT_OUTPUT_FILENAME = "trace.rpd"
+
+
+def benchmark_speculative_sampling(
```

Comment:
So you could also pull this out into a common utility and pass in the function to be profiled?